mono/metadata/marshal.c

```diff
@@ -345,13 +345,11 @@ mono_delegate_to_ftnptr_impl (MonoDelegateHandle delegate, MonoError *error)
 	}
 
 	if (method->flags & METHOD_ATTRIBUTE_PINVOKE_IMPL) {
-		const char *exc_class, *exc_arg;
 		gpointer ftnptr;
 
-		ftnptr = mono_lookup_pinvoke_call_internal (method, &exc_class, &exc_arg);
+		ftnptr = mono_lookup_pinvoke_call_internal (method, error);
 		if (!ftnptr) {
```

Comment:
Could we just use `if (!is_ok(error))`, or doesn't `mono_lookup_pinvoke_call_internal` guarantee that it will only return NULL in case of error?

Reply:
At the present, it doesn't have that guarantee.